src/coreclr/src/libraries-native/CMakeLists.txt

```diff
@@ -1,11 +1,22 @@
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
-set(GLOBALIZATION_NATIVE_DIR ${CLR_REPO_ROOT_DIR}/src/libraries/Native/Unix/System.Globalization.Native)
 
 # Suppress exporting of the PAL APIs
 add_definitions(-DPALEXPORT=EXTERN_C)
 
-include_directories("${GLOBALIZATION_NATIVE_DIR}")
-include_directories("${CLR_REPO_ROOT_DIR}/src/libraries/Native/Unix/Common")
+set(STATIC_LIBS_ONLY 1)
 
-add_subdirectory(${GLOBALIZATION_NATIVE_DIR} System.Globalization.Native)
+if(CLR_CMAKE_TARGET_WIN32)
```

Comment:
What does this pattern look like going forward? I would like to understand how this scales as we adopt more libraries. Questions like can someone take over and reasonably understand the design for the next static lib we want to fold into single file.

Reply:
The reason for odd pattern is historical. Ideally this whole file/folder should not be needed. (See further comment from @jkotas)

We should just `add_subdirectory(${CLR_REPO_ROOT_DIR}/src/libraries/Native/Unix Native.Unix)` directly from the parent.

We do not have platform symmetry here though and similar approach would not work on Windows as-is. 
`src/libraries/Native/Windows/CMakeLists.txt` needs some changes to make that possible.

Reply:
We no longer have "libraries-native" directory, so the pattern shoudl be more clear - we just build the same native libraries as libs partition does, although with a few different settings since we do not want DLLs and exports, but want static libraries instead.

Reply:
I think removal of "libraries-native" addresses the concern about the pattern here. This is not where we make the decision of what to link into the single file. We just build the native libs. For now we build all of them, since there is no reason to exclude.

Then the host build decides what to link. Presumably there could be more than one host that may link binaries with different set of embedded libraries. For now we have `singlefilehost` and in this change it links `System.Globalization` and `clrcompression`. (on Linux it also links other libs, but uses the old mechanism)

Once we are happy with these 2 libs, we will proceed to link others in the same way.